### PR TITLE
Dev fix login 

### PR DESCRIPTION
**Pull request by FrauDeborah**
Hi Johnathan,

To get the existing login tests to pass, I decided to convert the LoginDialogHost.Component from a route-able popup to a normal popup. That meant I needed to remove the LoginDialogHost.Service, and to touch the app. module, app.router, navigation, and the login components. Once that change was implemented to get the multiple existing tests to pass, I needed to add the provider values using spies for the LoginDialogHost.  I tell you that so you understand why multiple files were modified or deleted.

Interesting after all the changes, the LoginDialogHost.component looks more like a service than a component. Now I'm wondering if I should move remaining the functionality of the LoginDialogHost.component to the navigation.service and remove the LoginDialogHost.component. Alternatively, I could consider moving the LoginDialogHost.component functionaly to the Login.Service, but that might cause a circular path. Or we could leave well enough alone for now.

Was wondering your thoughts on the best solution. 

This PR works for both ng serve, and error free ng tests. Yet, it's not a good solution to have a component look like a service. If you merged the PR, I believe the ng test failure on dev would be resolved.

Let me know your thoughts.

PS. I'd like your feed back before I finish writing the more dialog tests. Also, this PR does not include the new functionally for the issue you groomed last sprint planning session. I can add the new method either before or after the new tests are written. Yet, I would like to use this PR as the basis for that new method. 

Deborah

### CODE REVIEW

File: src/app/_services/api.service.spec.ts [hunk 1]
```diff
@@ -88,28 +87,23 @@ describe('ApiService', () => {
...
 
-        req.flush({'name': true, 'mouseArray': [ 'Gray', 'Long tail', 'Squeaky' ],
-          'mouseData': mouseData
-        });
+        req.flush(returnedResult);
       })
   );
 
+
+/*
```
Comment (haxwell) [inline src/app/_services/api.service.spec.ts]: Just wondering why this is commented out.. the test passes, yes?

**Comment by FrauDeborah** [comment]
Hi Johnathan (@haxwell )

Thought I would let you know that I found an error. I forgot to change the login on the registration page. I'll be fixing the code and running more tests tonight. Sorry.

Deborah

File: src/app/_services/api.service.spec.ts (hunk 1 above)
Comment (FrauDeborah) [reply src/app/_services/api.service.spec.ts]: Johnathan (@haxwell )

When I was trying to determine why the tests were failing, I was attempting to reduce complexity by commenting out the tests. Let's say that I was confused at the time and forgot to add this test back in. I just un-commented the test; it ran with no issues. After I finish fixing the login from the registration page, I'll push the changes. I'm almost done, I just need to decide how to fix one more minor issue with the interaction of registration page with login. 

Deborah

**Comment by FrauDeborah** [comment]
Johnathan, @haxwell 

Recent change I pushed took the comment notation away from the apiService test and fixed the login on the registration page. I can't believe that the push previously neglected interaction of login and registration. Oh well!

I'm even more convinced that I need to refactor the LoginDialogHostComponent to make it a LoginDialogHostService or add the functionality into the LoginService. Why? Because the login dialog is launched from both the NavigationComponent and the RegistrationComponent now.

Any other feedback?

Deborah

**Pull request merged**